icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -154,13 +178,72 @@ void dumpEnumeratedProperty(UProperty uproperty, FILE* f) {
     fputs("[[enum_property]]\n", f);
     fprintf(f, "long_name = \"%s\"\n", fullPropName);
     if (shortPropName) fprintf(f, "short_name = \"%s\"\n", shortPropName);
+    dumpPropertyAliases(uproperty, f);
+
+    int32_t minValue = u_getIntPropertyMinValue(uproperty);
+    U_ASSERT(minValue >= 0);
+    int32_t maxValue = u_getIntPropertyMaxValue(uproperty);
+    U_ASSERT(maxValue >= 0);
+
+    std::vector<int> valueIntegers;
+    std::vector<const char*> longValueNames;
+    std::vector<const char*> shortValueNames;
+    std::vector<std::vector<const char*>> valueAliases;
+    for (int v = minValue; v <= maxValue; v++) {
+        const char* fullValueName = u_getPropertyValueName(uproperty, v, U_LONG_PROPERTY_NAME);
+        const char* shortValueName = u_getPropertyValueName(uproperty, v, U_SHORT_PROPERTY_NAME);
+        if (!fullValueName) {
+            continue;
+        }
+        valueIntegers.push_back(v);
+        longValueNames.push_back(fullValueName);
+        shortValueNames.push_back(shortValueName);
+
+        std::vector<const char*> aliases;
+        int i = U_LONG_PROPERTY_NAME + 1;
+        while(true) {
+            // The API works by having extra names after U_LONG_PROPERTY_NAME, sequentially,
+            // and returning null after that
+            const char* alias = u_getPropertyValueName(uproperty, v, (UPropertyNameChoice) i);
+            if (!alias) {
+                break;
+            }
+            aliases.push_back(alias);
+            i++;
+        }
+        valueAliases.push_back(aliases);
+    }
+
+
+    U_ASSERT(longValueNames.size() == valueIntegers.size());
```

Comment:
Suggestion: You don't need to collect these into a vector before printing them. You can print them in a single loop.

Reply:
That's true, I'm doing them all at once now.